x/src/cargo.rs

```diff
@@ -115,9 +128,15 @@ impl Cargo {
         self
     }
 
+    /// no fail fast will prevent propigation of errors.
     pub fn run(&mut self) -> Result<()> {
         self.inner.stdout(Stdio::inherit()).stderr(Stdio::inherit());
-        self.do_run(true).map(|_| ())
+        let output = self.do_run(true).map(|_| ());
+        if self.no_fail_fast {
+            Ok(())
+        } else {
+            output
+        }
```

Comment:
this is what I'm wondering about.   not entirely grasping the behavior of anyhow and ?.... appears to short circuit a function even if the result isn't inspected?